base/Enums.jl

```diff
@@ -21,6 +21,14 @@ Base.cconvert(::Type{T}, x::Enum{T2}) where {T<:Integer,T2<:Integer} = T(x)::T
 Base.write(io::IO, x::Enum{T}) where {T<:Integer} = write(io, T(x))
 Base.read(io::IO, ::Type{T}) where {T<:Enum} = T(read(io, basetype(T)))
 
+"""
+    _enum_hash(x::Enum, h::UInt)
+
+Compute hash for an enum value `x`. This internal method will be specialized
+for every enum type created through [`@enum`](@ref).
+"""
+_enum_hash(x::Enum, h::UInt) = hash(x, h)
+Base.hash(x::Enum, h::UInt) = _enum_hash(x, h)
```

Comment:
Can you add a unit test that includes a custom Enum with a custom-defined hash function? that should be a prerequisite before merging.

Reply:
It is not trivial to check for the warning, but I ended up adapting a similar test elsewhere.